tests/leak-tests/src/test/java/org/apache/activemq/artemis/tests/leak/ClientLeakTest.java

```diff
@@ -0,0 +1,137 @@
+         configuration.addAcceptorConfiguration(new TransportConfiguration(NETTY_ACCEPTOR_FACTORY, new HashMap<String, Object>(), "netty", new HashMap<String, Object>()));
+         ActiveMQServer server = ActiveMQServers.newActiveMQServer(configuration, false);
+         server.start();
+         System.out.println(LEAK_SERVER);
+      } catch (Throwable e) {
+         e.printStackTrace();
+         System.exit(-1);
+      }
+
+   }
+
+   @BeforeClass
+   public static void beforeClass() throws Exception {
+      Assume.assumeTrue(CheckLeak.isLoaded());
+   }
+
+   @Override
+   @Before
+   public void setUp() throws Exception {
+      serverProcess = SpawnedVMSupport.spawnVM(ClientLeakTest.class.getName());
+      runAfter(serverProcess::destroyForcibly);
+
+      boolean success = false;
+      long time = System.currentTimeMillis() + 5_000;
+
+      // this loop will keep trying a connection until the serer has started
+      do {
+         try {
+            ConnectionFactory cf = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:61616");
+            try (Connection connection = cf.createConnection()) {
+               success = true;
+            }
+         } catch (Throwable e) {
+            logger.debug(e.getMessage(), e);
+            Thread.sleep(100);
+         }
+
+      } while (success == false && System.currentTimeMillis() < time);
+      Assert.assertTrue(success);
+   }
+
+   @After
+   public void stopServer() throws Exception {
+      serverProcess.destroyForcibly();
+   }
+
+   @Test
+   public void testRepeatAMQPSessions() throws Exception {
+      CheckLeak checkLeak = new CheckLeak();
+
+      ConnectionFactory cf = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:61616");
+      Connection connection = cf.createConnection();
+      for (int i = 0; i < 10; i++) {
+         for (int j = 0; j < 10; j++) {
+            Session session = connection.createSession(true, Session.SESSION_TRANSACTED);
+            MessageProducer producer = session.createProducer(session.createQueue("test"));
+            producer.send(session.createTextMessage("test"));
+            session.commit();
+            session.close();
+         }
+
+         for (int j = 0; j < 10; j++) {
+            Session session = connection.createSession(true, Session.SESSION_TRANSACTED);
+            MessageConsumer consumer = session.createConsumer(session.createQueue("test"));
+            connection.start();
+            Message message = consumer.receive(1000);
+            Assert.assertNotNull(message);
+            session.commit();
+            // consumer.close(); // uncomment this and the test will pass.
```

Comment:
@gemmellr I am not sure if this has any relation to anything I'm seeing now... but if I remove this comment, the test will pass. closing the consumer will throw some log.warns on the server that I'm investigating and are expected to happen, but this will make the test to not keep any references on  any client Receivers.